MultiChannelDemo/Form1.cs

```diff
@@ -0,0 +1,450 @@
+                                               radioButtonSelectedPos5, radioButtonSelectedPos6, radioButtonSelectedPos7, radioButtonSelectedPos8 };
+
+            // initialize widgets
+            foreach (var gb in groupBoxes)
+                gb.Enabled = false;
+
+            initChart(chartAll);
+            foreach (var chart in charts)
+                initChart(chart);
+
+            clearSelection();
+
+            Text = String.Format("MultiChannelDemo v{0}", Application.ProductVersion);
+        }
+
+        void initChart(Chart chart)
+        {
+            var area = chart.ChartAreas[0];
+            area.AxisY.IsStartedFromZero = false;
+            area.AxisX.LabelStyle.Format = "{0:f2}";
+        }
+
+        void buttonInit_Click(object sender, EventArgs e)
+        {
+            if (initialized)
+                return;
+
+            if (!wrapper.open())
+            {
+                logger.error("failed to open MultiChannelWrapper");
+                return;
+            }
+
+            // per-channel initialization
+            chartAll.Series.Clear();
+            foreach (var pos in wrapper.positions)
+            {
+                var spec = wrapper.getSpectrometer(pos);
+                var sn = spec.serialNumber;
+
+                var gb = groupBoxes[pos - 1];
+                gb.Enabled = true;
+                gb.Text = $"Position {pos} ({sn})";
+
+                // big graph
+                var s = new Series($"Pos {pos} ({sn})");
+                s.ChartType = SeriesChartType.Line;
+                seriesAll[pos - 1] = s;
+                chartAll.Series.Add(s);
+            }
+
+            initialized = true;
+        }
+
+        protected override void OnFormClosing(FormClosingEventArgs e) => logger.setTextBox(null);
+
+        private void checkBoxVerbose_CheckedChanged(object sender, EventArgs e) => logger.level = checkBoxVerbose.Checked ? LogLevel.DEBUG : LogLevel.INFO;
+
+        ////////////////////////////////////////////////////////////////////////
+        // System-Level Control (all spectrometers)
+        ////////////////////////////////////////////////////////////////////////
+
+        // user clicked the "enable external hardware triggering" system checkbox 
+        void checkBoxTriggerEnableAll_CheckedChanged(object sender, EventArgs e) => wrapper.triggeringEnabled = checkBoxTriggerEnableAll.Checked;
+
+        // user clicked the "fan control" system checkbox
+        void checkBoxFanEnable_CheckedChanged(object sender, EventArgs e) => wrapper.fanEnabled = checkBoxFanEnable.Checked;
+
+        // user clicked the system-level "Acquire" button 
+        async void buttonAcquireAll_Click(object sender, EventArgs e)
+        {
+            enableControls(false);
+            await Task.Run(() => processSpectra(wrapper.getSpectra()));
+            enableControls(true);
+        }

```

Comment:
all this lightweight async await makes me jealous and anxious to carve out some time

Reply:
It proved a good time to learn it.  I'm just digging into Xamarin for Android, and realizing that the new async C# stuff is especially important on that platform.